src/extension.ts

```diff
@@ -18,22 +17,42 @@ function postMessage(panel: vscode.WebviewPanel, command: string, payload: objec
 	panel.webview.postMessage({ command: command, payload: payload });
 }
 
-function readCredentials(configuration: vscode.WorkspaceConfiguration, source: string, credentialsCallback: Function) {
+async function readCredentials(configuration: vscode.WorkspaceConfiguration, source: NugetSource): Promise<Credentials | undefined> {
+	if (!!source?.credentials) {
+		return source.credentials;
+	}
+
 	let command = "";
 	if (process.platform === 'win32') {
 		command = configuration.credentialProviderFolder + "/CredentialProvider.Microsoft.exe";
 	}
 	else {
 		command = "dotnet " + configuration.credentialProviderFolder + "/CredentialProvider.Microsoft.dll";
 	}
-	exec(command + " -C -F Json -U " + source, function callback(error: any, stdout: any, stderr: any) {
-		console.log(stderr)
-		credentialsCallback({ source: source, credentials: JSON.parse(stdout) });
-	});
+
+	return new Promise<Credentials | undefined>((resolve) => {
+		exec(command + " -C -F Json -U " + source.name, function callback(commandError: any, stdout: any, stderr: any) {
+			console.error(stderr)
+
+			if (!!commandError) {
+				console.error(`Error getting credentials for NuGet source ${source.name}`, commandError);
+				return resolve();
+			}
+
+			let credentials: Credentials | undefined;
+			try {
+				credentials = JSON.parse(stdout) as Credentials | undefined;
+			} catch (parserError) {
+				console.error(`Error getting credentials for NuGet source ${source.name}`, parserError);
+			}
+
+			resolve(credentials);
+		});
+	})
 }
 
 function loadProjects(panel: vscode.WebviewPanel) {
-	vscode.workspace.findFiles("**/*.csproj").then(files => {
+	vscode.workspace.findFiles("**/*.*proj").then(files => {
```

Comment:
Please, change it to support only `*.csproj` and  `*.fsproj`

Reply:
Will do. I would also include .vbproj so that all main three .NET languages are supported, ok?